### DISCUSSION TITLE
Some nodes do not get captured within quarto (qmd) or rmarkdown (Rmd) files, even though they appear in TSPlaygroundToggle.

### DISCUSSION THREAD

**Discussion by PMassicotte** [nvim-treesitter/nvim-treesitter#6160] [Q&A] [answered]
### Describe the highlighting problem

I know that quarto/rmd document (both based on markdown) do not have proper parser:

https://github.com/nvim-treesitter/nvim-treesitter/blob/d1484a1507b36ca6af7f7925c627846010faeabe/lua/nvim-treesitter/parsers.lua#L39

I am attempting to capture some nodes (like function definitions) but it is not working, even though they appear in TSPlaygroundToggle (see below).

### Example snippet that causes the problem

I have a simple quarto document with one R code chunk:

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/4519221/cea973c5-f7f2-4a66-b6d6-0bd6eb465e77)

I am able to capture some nodes, like headings and fenced code block:

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/4519221/259509d1-4046-4b08-a42a-9964bceab5d2)

However, I can not get any *code node* within the chunk: 

I am trying to capture code inside chunks (ex: *identifier*, *left_assignment*, etc.):

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/4519221/f48f68ea-fc65-46e1-9590-4ea420a9336c)

I am wondering if this is expected, as they are correctly identified in the syntax tree (see attached screenshots).

### Tree-sitter parsing result

```
section [0, 0] - [13, 0]
  atx_heading [0, 0] - [1, 0]
    atx_h1_marker [0, 0] - [0, 1]
    heading_content: inline [0, 2] - [0, 11]
  section [2, 0] - [13, 0]
    atx_heading [2, 0] - [3, 0]
      atx_h2_marker [2, 0] - [2, 2]
      heading_content: inline [2, 3] - [2, 12]
    fenced_code_block [4, 0] - [13, 0]
      fenced_code_block_delimiter [4, 0] - [4, 3]
      info_string [4, 3] - [4, 6]
        language [4, 4] - [4, 5]
      block_continuation [5, 0] - [5, 0]
      code_fence_content [5, 0] - [12, 0]
        left_assignment [5, 0] - [5, 6]
          name: identifier [5, 0] - [5, 1]
          value: float [5, 5] - [5, 6]
        left_assignment [7, 0] - [9, 1]
          name: identifier [7, 0] - [7, 7]
          value: function_definition [7, 11] - [9, 1]
            formal_parameters [7, 19] - [7, 22]
              identifier [7, 20] - [7, 21]
            brace_list [7, 23] - [9, 1]
              binary [8, 2] - [8, 7]
                left: identifier [8, 2] - [8, 3]
                right: float [8, 6] - [8, 7]
        call [11, 0] - [11, 10]
          function: identifier [11, 0] - [11, 7]
          arguments: arguments [11, 8] - [11, 9]
            identifier [11, 8] - [11, 9]
        block_continuation [6, 0] - [6, 0]
        block_continuation [7, 0] - [7, 0]
        block_continuation [8, 0] - [8, 0]
        block_continuation [9, 0] - [9, 0]
        block_continuation [10, 0] - [10, 0]
        block_continuation [11, 0] - [11, 0]
        block_continuation [12, 0] - [12, 0]
      fenced_code_block_delimiter [12, 0] - [12, 3]
```

### Example screenshot

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/4519221/66cb7e9e-2d58-4753-aeef-c9d668451bb2)

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check() + 

Installation ~
- OK `tree-sitter` found 0.20.8 (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v16.20.2 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.5.0-17-generic",
  sysname = "Linux",
  version = "#17~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Jan 16 14:32:32 UTC 2"
} ~

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - cmake               ✓ . ✓ ✓ .
  - cpp                 ✓ ✓ ✓ ✓ ✓
  - css                 ✓ . ✓ ✓ ✓
  - csv                 ✓ . . . .
  - diff                ✓ . . . .
  - go                  ✓ ✓ ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - json                ✓ ✓ ✓ ✓ .
  - julia               ✓ ✓ ✓ ✓ ✓
  - latex               ✓ . ✓ . ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - luadoc              ✓ . . . .
  - luap                ✓ . . . .
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - r                   ✓ ✓ . ✓ ✓
  - regex               ✓ . . . .
  - ron                 ✓ ✓ ✓ ✓ ✓
  - rust                ✓ ✓ ✓ ✓ ✓
  - sql                 ✓ . . ✓ ✓
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

```
```


### Output of `nvim --version`

```text
NVIM v0.10.0-dev-2034+g6635ec113
Build type: RelWithDebInfo
LuaJIT 2.1.1703358377
Run "nvim -V1 -v" for more info
```

**Comment by clason** [comment]
The markdown parser is special; some nodes are provided only by the `markdown_inline` parser.

**Reply by clason** [reply]
And the R nodes cannot be captured by a `markdown` query; you need to do this in an R query. (The playground is deprecated, by the way. Use `:Inspect`/`:InspectTree`/`:EditQuery` instead.)

**Comment by PMassicotte** [comment]
Thank you for your responses. Here is an updated screenshot with `:Inspect`

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/4519221/aa6f330d-cea2-441b-acd5-04a87d100918)

It now provides useful information, like telling that I am trying to capture an invalid node (maybe update the issue template that suggest using the old playground).

Can you give me some hints when you say that I have to do it in an R query? At the moment, I am working on a neovim plugin in lua.

**Reply by clason** [reply, marked as answer]
use `I` to toggle the language. Every query is specific to a language (parser). The markdown parser only knows about the markdown structure; Neovim tells tree-sitter to parse codeblocks marked as `{r}` with the R parser (and queries); this is called "injection".

Look at the queries in this repo.

**Comment by PMassicotte** [comment]
Thank you very much for your help. I will look at this.
Cheers and thank again.
